supplier_api_tester/supplier_api_tester/tests/availability_timeslots.py

```diff
@@ -1,39 +1,96 @@
-from datetime import datetime, timedelta
-from typing import List
+# If multiple checks can be done using the same response
+# then they should be done under a single test case.
 
-import requests
+from datetime import datetime, timedelta
 
-from ..client import client
 from ..decorators import test_wrapper
 from ..exceptions import FailedTest
-from ..models import TestResult, ApiError
+from ..models import TestResult
 from ..utils import availability
-from ..utils.adapters import get_timeslots
-
-
-@test_wrapper
-def test_response_format(api_url, api_key, product_id, version=1):
-    '''[Timeslots] Checking response format'''
-    return availability.test_response_format(api_url, api_key, product_id, 'timeslots', get_timeslots, version=1)
 
 
 @test_wrapper
 def test_next_30_days(api_url, api_key, product_id, version=1):
-    '''[Timeslots] Checking for any availability in the next 30 days'''
-    return availability.test_next_30_days(api_url, api_key, product_id, 'timeslots', get_timeslots, version=1)
+    '''[Timeslots] Checking availability for the next 30 days'''
+    start_date = datetime.utcnow().date()
+    end_date = start_date + timedelta(days=30)
+    timeslots, raw_response = availability.get_availability_timeslots(
+        api_url,
+        api_key,
+        product_id,
+        version,
+        start_date,
+        end_date,
+    )
+
+    # non-zero availability check
+    max_tickets_sum = sum(timeslot.max_tickets for timeslot in timeslots)
```

Comment:
I think this along with variant uniquness is duplicated in 30 days check. Is it worth to extract this?

Reply:
I left it only to show that we're iterating over different objects.